addons/website_event_track/static/src/js/website_event_track_proposal_form.js

```diff
@@ -0,0 +1,175 @@
+    this._updateErrorDisplay(formErrors);
+    return formErrors.length === 0;
+},
+
+/**
+ * If there are still errors in form, display the error section and
+ * compose the error message accordingly.
+ * 
+ * @private
+ * @param {Array} errors - Names of errors still present in form.
+ */
+_updateErrorDisplay: function(errors) {
+
+    if (!errors.length) {
+        this.$('.o_wetrack_proposal_error_section').addClass('d-none');
+    } else { 
+        this.$('.o_wetrack_proposal_error_section').removeClass('d-none');
+    }
+
+    var errorMessage = '';
+    var $errorElement = this.$('.o_wetrack_proposal_error_message');
+
+    if (errors.includes('invalidFormInputs')) { errorMessage += _t(' Please fill out the form correctly.'); }
+    if (errors.includes('noContactMean')) { errorMessage += _t(' Please enter a contact email address or a contact phone number.'); }
+    if (errors.includes('submitError')) { errorMessage += _t(' An error has occured, the form has not been sent.'); }
+
+    $errorElement.text(errorMessage).change();
+},
+
+//--------------------------------------------------------------------------
+// Handlers
+//--------------------------------------------------------------------------
+
+/**
+ * Display / Hide Additional Contact Information section when toggling
+ * the checkbox on the form o_wetrack_add_contact_information_checkbox.
+ * Also empty the email to prevent hidden email format error.
+ * 
+ * @private
+ * @param {Event} ev
+ */
+_onAdvancedContactToggle: function(ev) {
+    this.useAdvancedContact = !this.useAdvancedContact;
+    var $contactName = this.$(".o_wetrack_contact_name_input")[0];
+    var $advancedInformation = this.$('.o_wetrack_contact_information');
+
+    if (this.useAdvancedContact) {
+        $advancedInformation.removeClass('d-none');
+        $contactName.setAttribute("required", "True");
+    } else {
+        this.$('.o_wetrack_contact_email_input').val('').change();
+        $advancedInformation.addClass('d-none');
+        $contactName.removeAttribute("required");
+    }
+},
+
+/**
+ * Propagates the speaker name to contact name on new input. 
+ * 
+ * @private
+ * @param {Event} ev
+ */
+_onPartnerNameInput: function(ev) {
+    var text = $(ev.currentTarget).val();
+    this.$(".o_wetrack_contact_name_input").val(text).change();
+},
+
+/**
+ * Submits the form if no errors are present in the form after validation.
+ * 
+ * @private
+ * @param {Event} ev
+ */
+_onProposalFormSubmit: function(ev) {
+    ev.preventDefault();
+    ev.stopPropagation();
+
+    // Prevent further clicking
+    this.$target.find('.o_wetrack_proposal_submit_button')
+        .addClass('disabled')
+        .attr('disabled', 'disabled');
+
+    // Submission of the form if no errors remain
+    if (this._isFormValid()) { 
+        try {
+            this.$el.submit();
+        } catch(e) {
+            this._updateErrorDisplay(['submitError']);
+        }
+        this.$target[0].reset();
+    }
+
```

Comment:
Here I just added a reset of the form. This also prevents weird behavior coming back to the page after a python Exception has been raised.